velox/substrait/tests/VeloxSubstraitRoundTripPlanConverterTest.cpp

```diff
@@ -209,6 +209,71 @@ TEST_F(VeloxSubstraitRoundTripPlanConverterTest, sumMask) {
       "FROM tmp");
 }
 
+TEST_F(VeloxSubstraitRoundTripPlanConverterTest, rowConstructor) {
+  RowVectorPtr vectors = makeRowVector(
+      {makeFlatVector<double_t>({0.905791934145, 0.968867771124}),
+       makeFlatVector<int64_t>({2499109626526694126, 2342493223442167775}),
+       makeFlatVector<int32_t>({581869302, -133711905})});
+  createDuckDbTable({vectors});
+
+  auto plan = PlanBuilder()
+                  .values({vectors})
+                  .project({"row_constructor(c1, c2)"})
+                  .planNode();
+  assertPlanConversion(plan, "SELECT row(c1, c2) FROM tmp");
+}
+
+TEST_F(VeloxSubstraitRoundTripPlanConverterTest, projectAs) {
+  RowVectorPtr vectors = makeRowVector(
+      {makeFlatVector<double_t>({0.905791934145, 0.968867771124}),
+       makeFlatVector<int64_t>({2499109626526694126, 2342493223442167775}),
+       makeFlatVector<int32_t>({581869302, -133711905})});
+  createDuckDbTable({vectors});
+
+  auto plan = PlanBuilder()
+                  .values({vectors})
+                  .filter("c0 < 0.5")
+                  .project({"c1 * c2 as revenue"})
+                  .partialAggregation({}, {"sum(revenue)"})
+                  .planNode();
+  assertPlanConversion(
+      plan, "SELECT sum(c1 * c2) as revenue FROM tmp WHERE c0 < 0.5");
+}
+
+TEST_F(VeloxSubstraitRoundTripPlanConverterTest, avg) {
+  auto vectors = makeVectors(2, 7, 3);
+  createDuckDbTable(vectors);
+
+  auto plan = PlanBuilder()
+                  .values(vectors)
+                  .partialAggregation({}, {"avg(c4)"})
+                  .finalAggregation()
+                  .planNode();
+
+  assertPlanConversion(plan, "SELECT avg(c4) FROM tmp");
+}
+
+TEST_F(VeloxSubstraitRoundTripPlanConverterTest, rowConstructorAgg) {
+  std::vector<RowVectorPtr> vectors;
+  auto vector = makeRowVector({
+      makeFlatVector<double>(
+          1'000, [](auto row) { return row % 11; }, nullEvery(3)),
+      makeFlatVector<double>(
+          1'000, [](auto row) { return row; }, nullEvery(2)),
+  });
+  vectors.push_back(vector);
+  createDuckDbTable(vectors);
+  auto plan =
+      PlanBuilder()
+          .values(vectors)
+          .partialAggregation(
+              {}, {"sum(c1) as c1_sum", "count(c1) as c1_count"})
+          .project({"row_constructor(c1_sum, c1_count) AS c1_sum_count"})
```

Comment:
As discussed in #2385, this test is not valid. Final aggregation should receive output of partial aggregation unmodified.

Reply:
Yes, this test cannot pass if the testing data contains null. We can remove it since we plan to support struct type in columnar shuffle.